### ISSUE TITLE
Skip building C extensions for the target environement

### ISSUE THREAD

**Issue by rth** [pyodide/pyodide#230]
Initially I though that when we build  C extensions for Python modules, we have a first pass with gcc that generates the intermediary representation (`.bc` files) that are than cross-compiled with `emcc` to `.wasm`.

However it looks like it is not the case and `.bc` files are generated directly by `emcc`? In which case the questions is whether it is useful to actually build the C extensions for target as that is responsible of probably around ~40% of build time for numpy, pandas, scipy etc. Can we skip that step altogether which would help for https://github.com/iodide-project/pyodide/issues/76 ? 

Though maybe I'm missing something  about how cross-compiling works in `pyodide_build` @mdboom..

**Comment by mdboom** [comment]
It's done that way primarily because there is no other way that I've found to get out the list of steps needed for compilation.  If you do something that doesn't actually produce files, distutils will stop compilation, but it might be possible to just create empty `.o` and `.so` files (and not actually call `gcc`) in the first pass which would speed things up considerably.

In earlier versions of Pyodide, I did this by hand and converted each package to Makefiles.  We could "record" once and store the build.log files in the repo, of course.  That's a little farther from how things may eventually work in conda-build down the road, though.
